<h1>Activity 1 - Palmer Penguin Data Analytics</h1>
<hr>
<h3>Analyst: John Oliver Liwanag</h3>

In [2]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing dataset to Jupyter notebook </p>
<p>Converting dataset to dataframe</p>
<ul>Requirements
    <li>Name of the dataframe</li>
    <li>Source</li>
</ul>

In [4]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [5]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [6]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [7]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Cleaning the Data Frame</h1>

In [9]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Selecting a column</h2>
<h3>Syntax</h3><br>
data_frame['column_name']

In [11]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [12]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [13]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [14]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [15]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [16]:
penguins['body_mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  body_mass_kg  
0         3750.0    male  2007           3.8  
1         3800.0  female  2007           3.8  
2         3250.0  female  2007           3.2  
3            NaN     NaN  2007           NaN  
4         3450.0  female  2007           3.4  
..           ...     ...   ...           ...  
339       4000.0    male  2009           4.0  
340       3400.0  female  2009           3.4  
341       3775.0    male  2009           3.8  
342       4100.0    male  2009           4.1  
343       3775.0  female  2009           3.8  

[344 rows x 9 columns]

In [17]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  body_mass_kg  
0         3750.0    male  2007           3.8  
1         3800.0  female  2007           3.8  
2         3250.0  female  2007           3.2  
3            NaN     NaN  2007           NaN  
4         3450.0  female  2007           3.4  
..           ...     ...   ...           ...  
147       3475.0  female  2009           3.5  
148       3450.0  female  2009           3.4  
149       3750.0    male  2009           3.8  
150       3700.0  female  2009           3.7  
151       4000.0    male  2009           4.0  

[152 rows x 9 columns]

In [18]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  body_mass_kg  
152       4500.0  female  2007           4.5  
153       5700.0    male  2007           5.7  
154       4450.0  female  2007           4.4  
155       5700.0    male  2007           5.7  
156       5400.0    male  2007           5.4  
..           ...     ...   ...           ...  
271          NaN     NaN  2009           NaN  
272       4850.0  female  2009           4.8  
273       5750.0    male  2009           5.8  
274       5200.0  female  2009           5.2  
275       5400.0    male  2009           5.4  

[124 rows x 9 columns]

In [19]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  body_mass_kg  
276       3500.0  female  2007           3.5  
277       3900.0    male  2007           3.9  
278       3650.0    male  2007           3.6  
279       3525.0  female  2007           3.5  
280       3725.0    male  2007           3.7  
..           ...     ...   ...           ...  
339       4000.0    male  2009           4.0  
340       3400.0  female  2009           3.4  
341       3775.0    male  2009           3.8  
342       4100.0    male  2009           4.1  
343       3775.0  female  2009           3.8  

[68 rows x 9 columns]

In [20]:
(penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [21]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [22]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  body_mass_kg  
284       4150.0  female  2007           4.2  
289       4050.0    male  2007           4.0  
291       4050.0    male  2007           4.0  
295       4400.0    male  2007           4.4  
301       4150.0    male  2007           4.2  
305       4550.0    male  2008           4.6  
307       4300.0    male  2008           4.3  
309       4100.0    male  2008           4.1  
313       4800.0    male  2008           4.8  
315       4500.0    male  2008           4.5  
321       4450.0    male  2009           4.4  
323       4300.0    male  2009           4.3  
329       4050.0    male  2009           4.0  
333       4050.0    male  2009           4.0  
339       4000.0    male  2009           4.0  
342       4100.0    male  2009           4.1

In [23]:
chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]
chinstrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
 8   body_mass_kg       16 non-null     float64
dtypes: float64(5), int64(1), object(3)
memory usage: 1.2+ KB


<h1>How to Fill Null Values</h1>

In [25]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
body_mass_kg          2
dtype: int64

In [26]:
penguins['bill_length_mm'].mean()

43.9219298245614

In [27]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
body_mass_kg          2
dtype: int64

In [28]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [29]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
body_mass_kg         2
dtype: int64

In [30]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
body_mass_kg         2
dtype: int64

In [31]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          2
sex                  0
year                 0
body_mass_kg         2
dtype: int64

In [32]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
body_mass_kg         2
dtype: int64

In [33]:
penguins['body_mass_kg'] = penguins['body_mass_g'].fillna(penguins['body_mass_kg'].mean())
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
body_mass_kg         0
dtype: int64

<h1>How to Create a Column</h1>

In [35]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year', 'body_mass_kg'],
      dtype='object')

In [36]:
# penguins['body_mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)
# penguins

<h1>How to Change the Data Type of a Column</h1>

In [38]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
 8   body_mass_kg       344 non-null    float64
dtypes: float64(5), int64(1), object(3)
memory usage: 24.3+ KB


In [39]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [40]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
 8   body_mass_kg       344 non-null    float64 
dtypes: category(3), float64(5), int64(1)
memory usage: 17.6 KB


<h1>Data Aggregation</h1>
<h3>1. Species per island</h3>

In [42]:
island = penguins.groupby(['species', 'island']).count()
island

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  body_mass_kg  
species   island                                           
Adelie    Biscoe              44   44    44            44  
          Dream               56   56    56            56  
          Torgersen           52   52    52            52  
Chinstrap Biscoe               0    0     0             0  
          Dream               68   68    68            68  
          Torgersen            0    0     0             0  
Gentoo    Biscoe             124  124   124           124  
          Dream                0    0     0             0  
          Torgersen            0    0     0             0

In [43]:
island = penguins.groupby(['species', 'sex']).size()
island

species    sex   
Adelie     female    73
           male      79
Chinstrap  female    34
           male      34
Gentoo     female    58
           male      66
dtype: int64

In [44]:
island = penguins.groupby(['species', 'island', 'sex']).size()
island

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

In [45]:
avg_species_island = penguins.groupby(['species', 'island'])['body_mass_kg'].mean()
avg_species_island

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3715.899123
Chinstrap  Biscoe               NaN
           Dream        3733.088235
           Torgersen            NaN
Gentoo     Biscoe       5068.965761
           Dream                NaN
           Torgersen            NaN
Name: body_mass_kg, dtype: float64

<h1>Testing Correlation</h1>
<h3>Pearson R Correlation</h3>
<hr>
<h3>Use for testing between two <strong>NUMERICAL VALUES</h3>

In [47]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])
weight_flipper_cor

0.8712017673060115

<h1>Exploratory Data Analysis (EDA)</h1>
<h3>1. Which species has the highest/heaviest body mass in kg?</h3>

In [49]:
chinstrap_df['body_mass_kg'].mean()

3.730882352941177

In [50]:
adelie_df['body_mass_kg'].mean()

3.7019867549668874

In [51]:
gentoo_df['body_mass_kg'].mean()

5.076422764227642

<h1>Insight #1</h1>
<h3>The heaviest species from the dataset is the Gentoo species.</h3>
<hr>
<h3>2. Which species has more males than females?</h3>

In [53]:
chinstrap_df['sex'].mode()

0    female
1      male
Name: sex, dtype: object

In [54]:
adelie_df['sex'].mode()

0    female
1      male
Name: sex, dtype: object

In [55]:
gentoo_df['sex'].mode()

0    male
Name: sex, dtype: object

<h1>Insight #2</h1>
<h3>The only species with more males than females is the Gentoo Species.</h3>
<hr>
<h3>3. What species has the most penguins in 2009?</h3>

In [57]:
chinstrap_df['year'].value_counts()

year
2007    26
2009    24
2008    18
Name: count, dtype: int64

In [58]:
adelie_df['year'].value_counts()

year
2009    52
2007    50
2008    50
Name: count, dtype: int64

In [59]:
gentoo_df['year'].value_counts()

year
2008    46
2009    44
2007    34
Name: count, dtype: int64

<h1>Insight #3</h1>
<h3>Adelie has the most in the year 2009 with 52 penguins.</h3>
<hr>
<h3>4. How many penguins have above 190mm in flipper length?</h3>

In [61]:
(penguins['flipper_length_mm'] >= 190).value_counts()

flipper_length_mm
True     267
False     77
Name: count, dtype: int64

<h1>Insight #4</h1>
<h3>There are about 267 penguins that have 190mm or above in flipper length.</h3>
<hr>
<h3>5. What is the highest bill depth in mm with the Chinstrap species?</h3>

In [63]:
chinstrap_df.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm      68.0    48.833824    3.339256    40.9    46.350    49.55   
bill_depth_mm       68.0    18.420588    1.135395    16.4    17.500    18.45   
flipper_length_mm   68.0   195.823529    7.131894   178.0   191.000   196.00   
body_mass_g         68.0  3733.088235  384.335081  2700.0  3487.500  3700.00   
year                68.0  2007.970588    0.863360  2007.0  2007.000  2008.00   
body_mass_kg        68.0     3.730882    0.389894     2.7     3.475     3.70   

                        75%     max  
bill_length_mm       51.075    58.0  
bill_depth_mm        19.400    20.8  
flipper_length_mm   201.000   212.0  
body_mass_g        3950.000  4800.0  
year               2009.000  2009.0  
body_mass_kg          4.000     4.8

In [64]:
chinstrap_df['bill_depth_mm'].max()

20.8

<h1>Insight #5</h1>
<h3>The highest bill depth with the Chinstrap species is about 20.8mm</h3>
<hr>
<h3>6. What year has the lowest/shortest bill length?</h3>

In [66]:
df_2007 = penguins[penguins['year'] == 2007]
df_2008 = penguins[penguins['year'] == 2008]
df_2009 = penguins[penguins['year'] == 2009]

In [67]:
df_2007['bill_length_mm'].mean()

43.74201754385965

In [68]:
df_2008['bill_length_mm'].mean()

43.541228070175436

In [69]:
df_2009['bill_length_mm'].mean()

44.44851608187135

<h1>Insight #6</h1>
<h3>2009 have the highest bill length with an average of 44.44</h3>
<hr>
<h3>7. What year has the highest/longest flipper length with the Adelie species?</h3>

In [71]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  body_mass_kg  
0         3750.0    male  2007           3.8  
1         3800.0  female  2007           3.8  
2         3250.0  female  2007           3.2  
3            NaN     NaN  2007           NaN  
4         3450.0  female  2007           3.4  
..           ...     ...   ...           ...  
147       3475.0  female  2009           3.5  
148       3450.0  female  2009           3.4  
149       3750.0    male  2009           3.8  
150       3700.0  female  2009           3.7  
151       4000.0    male  2009           4.0  

[152 rows x 9 columns]

In [72]:
df_2007[df_2007['species'] == 'Adelie']['flipper_length_mm'].mean()

186.87830409356724

In [73]:
df_2008[df_2008['species'] == 'Adelie']['flipper_length_mm'].mean()

191.04

In [74]:
df_2009[df_2009['species'] == 'Adelie']['flipper_length_mm'].mean()

192.07692307692307

<h1>Insight #7</h1>
<h3>2009 have the highest flipper length of the Adelie species.</h3>
<hr>

<h1>Data Aggregation</h1>
<h3>8. Which species have the highest flipper length in 2009?</h3>

In [77]:
avg_species_island = penguins.groupby(['species', 'year'])['flipper_length_mm'].mean()
avg_species_island

species    year
Adelie     2007    186.878304
           2008    191.040000
           2009    192.076923
Chinstrap  2007    192.423077
           2008    197.722222
           2009    198.083333
Gentoo     2007    215.117647
           2008    217.565217
           2009    218.020800
Name: flipper_length_mm, dtype: float64

<h1>Insight #8</h1>
<h3>The species with the highest flipper length in 2009 is the Gentoo species with an average of 218.02.</h3>
<hr>
<h3>9. What island has more male than female from Adelie species?</h3>

In [79]:
island = penguins.groupby(['island', 'species', 'sex']).size()
island

island     species    sex   
Biscoe     Adelie     female    22
                      male      22
           Chinstrap  female     0
                      male       0
           Gentoo     female    58
                      male      66
Dream      Adelie     female    27
                      male      29
           Chinstrap  female    34
                      male      34
           Gentoo     female     0
                      male       0
Torgersen  Adelie     female    24
                      male      28
           Chinstrap  female     0
                      male       0
           Gentoo     female     0
                      male       0
dtype: int64

<h1>Insight #9</h1>
<h3>The island Dream have more male than female from Adelie species.</h3>
<hr>
<h3>10. How many females from the Chinstrap species have a body mass lower than 4kg?</h3>

In [81]:
island = penguins[penguins['body_mass_g'] <= 4000].groupby(['species', 'sex']).size()
island

species    sex   
Adelie     female    73
           male      43
Chinstrap  female    33
           male      20
Gentoo     female     1
           male       0
dtype: int64

<h1>Insight #10</h1>
<h3>There are 33 females from the chinstrap species with a body mass lower than 4kg.</h3>
<hr>
<h1>Data Correlation</h1>
<h3>11. Is there a correlation between the body mass and bill length?</h3>

In [83]:
weight_length_cor = penguins['body_mass_kg'].corr(penguins['bill_length_mm'])
weight_length_cor

0.59510982443763

<h1>Insight #11</h1>
<h3>There is a correlation between the body mass and bill length.</h3>
<hr>
<h3>12. Is there a correlation between the body mass and bill length?</h3>

In [85]:
bill_flipper_cor = penguins['bill_length_mm'].corr(penguins['flipper_length_mm'])
bill_flipper_cor

0.6561813407464281

<h1>Insight #12</h1>
<h3>There is a correlation between the bill length and flipper length.</h3>
<hr>